core/src/finagle_clojure/util/test_helpers.clj

```diff
@@ -0,0 +1,15 @@
+(ns finagle-clojure.util.test-helpers
+  (:require [midje.sweet :refer :all]
+            [midje.checking.core :as checking]
+            [finagle-clojure.futures :as fut])
+  (:import (com.twitter.util ConstFuture Promise)))
+
+(defchecker future= [expected]
+            (checker [actual]
+                     (if-not (or (instance? Promise actual) (instance? ConstFuture actual))
+                       (checking/as-data-laden-falsehood {:notes [(str "Expected a Future/Promise object but got a " (class actual) ": " actual)]})
+                       (let [actual-value (fut/await actual)]
```

Comment:
I'd encourage you to add a timeout here so your test doesn't hang if the promise is never satisfied.